server/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -359,7 +378,8 @@ default void restore() {
         private final Map<String, Object> transientHeaders;
         private final Map<String, List<String>> responseHeaders;
         private final boolean isSystemContext;
-
+        private long wrnHeadersSize; //saving current warning headers' size not to recalculate the size with every new warning header
+        private boolean isWrnLmtReached;
```

Comment:
Is this field strictly needed?

Reply:
I think yes. this field once the limit is reached ensures that we don't check and log "warnings limit reached" again and again with a new warning header.